src/main/perl/Blockdevices.pm

```diff
@@ -105,6 +107,272 @@ sub should_create_ks
+
+Compute the smallest interval for the expected C<size>
+given the conditions (one or more of):
+
+=over
+
+=item fraction : double
+
+The difference between the found and the expected size is at most 
+C<fraction> of the expected size. C<fraction> is a double 
+(e.g. 1 percent is 0.01).
+
+=item diff : long
+
+The difference between the found and the expected size (in MiB) is at most 
+C<diff> MiB.
+
+=back
+
+=cut
+
+sub correct_size_interval
+{
+    my $self = shift;
+
+    my @conds = sort keys %{$self->{correct}->{size}};
+    $this_app->verbose("Going to use ", scalar @conds, " conditions: ", join(", ", @conds));    
+
+    my ($min, $max);
+
+    # Update min and max with possible new minimum and maximum.
+    my $update_min_max = sub {
+        my ($nmin, $nmax) = @_;
+        if (defined($min)) {
+            $min = $nmin if ($nmin > $min);
+        } else {
+            $min = $nmin;
+        }
+        if (defined($max)) {
+            $max = $nmax if ($nmax < $max);
+        } else {
+            $max = $nmax;
+        }
+    };
+
+    foreach my $cond (@conds) {
+        if ($cond eq "diff") {
+            my $diff =  $self->{correct}->{size}->{diff};
+            $update_min_max->($self->{size} - $diff, $self->{size} + $diff);
+            $this_app->verbose("Diff defined $diff, updated min/max $min / $max");
+        } elsif ($cond eq "fraction") {
+            my $frac = $self->{correct}->{size}->{fraction};
+            $update_min_max->((1-$frac) * $self->{size}, (1+$frac) * $self->{size});
+            $this_app->verbose("Fraction defined $frac, updated min/max $min / $max");
+        } else {
+            $this_app->error("is_correct_size unknown condition $cond");
+            return;
+        }
+    };
+
+
+    if(!defined($min)) {
+        $this_app->info("Minimum undefined after the conditions, using expected size $self->{size}");
+        $min = $self->{size};
+    }
+    
+    if(!defined($max)) {
+        $this_app->info("Maximum undefined after the conditions, using expected size $self->{size}");
+        $max = $self->{size};
+    }
+
+    $min = 0 if ($min < 0);
+    $max = 0 if ($max < 0);
+
+    if($min > $max) {
+        $this_app->error("is_correct_size minimum $min larger then maximum $max");
+        return;
+    }
+
+
+    return ($min, $max);
+}
+
+=pod
+
+=head2 is_correct_size
+
+Returns true if the size of this device
+lies in the C<correct_size_interval>.
+
+Returns undef if device does not exists.
+
+Logs error if attributes are missing (like C<size>), 
+possibly due to incomplete profiles.
+
+=cut
+
+sub is_correct_size
+{
+    my $self = shift;
+    
+    if(! defined($self->{size})) {
+        $this_app->error("size attribute not defined (missing in profile?)");
+        # considered failed. don't specify "correct/size" if you don't want this to run?
+        return 0;
+    }
+
+    if(! $self->{correct}->{size}) {
+        $this_app->error("correct/size not defined (missing in profile?)");
```

Comment:
the method throws an error because it is expecting the sub-path `correct/size` (from something like
```pan
"/system/blockdevices/physical_devs/sdb/correct/size/diff" = 100;
```
)